Frontend/src/app/components/banner/CookieConsent.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+
+import { CookieIcon } from "lucide-react";
+import { Button } from "@/components/ui/button";
+import { useEffect, useState } from "react";
+import { cn } from "@/lib/utils";
+
+import Link from "next/link";
+import { setCookiesAccepted } from "@/lib/cookies";
+import {
+  deleteSessionFromCookies,
+  commitSessionToCookies,
+} from "@/lib/session";
+
+export default function CookieConsent({
+  variant = "default",
+  onAcceptCallback = async () => {
+    await setCookiesAccepted();
+    await commitSessionToCookies();
+  },
+  onDeclineCallback = async () => {
+    try {
+      await deleteSessionFromCookies();
+      console.log("User denied cookie consent");
```

Comment:
It's not necessary to log this to console

Reply:
Deleted